module/mempool/stdmap/pending_receipts.go

```diff
@@ -52,18 +52,18 @@ func NewPendingReceipts(headers storage.Headers, limit uint) *PendingReceipts {
 	// See https://github.com/onflow/flow-go/pull/387/files#r574228078
 	r.RegisterEjectionCallbacks(func(entity flow.Entity) {
 		receipt := entity.(*flow.ExecutionReceipt)
-		removeReceipt(receipt, r.entities, r.byPreviousResultID)
+		removeReceipt(receipt, r.backData, r.byPreviousResultID)
 	})
 	return r
 }
 
 func removeReceipt(
 	receipt *flow.ExecutionReceipt,
-	entities map[flow.Identifier]flow.Entity,
+	entities mempool.BackData,
```

Comment:
refactoring `entities` from a `map`-based structure to an `interface` that supports both `map` and `array` implementations.